### ISSUE TITLE
Feature Request: Nested Templates

### ISSUE THREAD

**Issue by mattdb** [palkan/rbytes#2]
## Is your feature request related to a problem? Please describe.

I'd like to be able to incorporate existing templates with my own modifications to them, from a single repository. For example, I'd like to have a "supertemplate" for Rails that installs a few default gems, etc, and then launches a couple of other templates that I've modified (Ruby on Whales, RuboCoping). I can commingle all of the templates from both sub-templates in my templates directory, but it immediately becomes quite messy, which template file came from which of the sub-templates (or the "supertemplate") is unclear from the directory structure alone.

## Describe the solution you'd like

I'd like to be able to store sub-template files in a subdirectory, then specify in my template something like:
```
<%= sub_template('ruby-on-whales', 'ruby-on-whales') %>
```
Where the first argument is the subdirectory to consider the new "root" while rendering that template, the second being the template within that directory to start rendering (though having a default of a matching name would be a bonus)

alternately, something like Thor's `inside`:
```
<%= inside('ruby-on-whales') do %>
  <%= include 'ruby-on-whales' %>
<% end %>
```

## Describe alternatives you've considered

I _could_ accomplish most of this by forking the different RBytes repos that I wish to modify, publishing them, and then including them into the "supertemplate" with `run "rails app:template LOCATION=..."` , however, this requires a lot of moving parts and is hard to iterate on, and spreads our custom configurations across multiple repos (forks + "supertemplate").

I _think_ the `sub_template` command could be accomplished by adding an additional method to `RubyBytes::Compiler` that would instantiate a new `Compiler` with the new root, capture the output, and include that in the template at that point. I think there's also a way to use a stack of paths similar to the way Thor implements `inside`, which would potentially avoid multiple `Compiler` instances and passing huge strings around.

## Additional context

Happy to contribute a PR if this seems like something of use to you - let me know which approach you'd prefer (or if there's another way entirely that seems worth exploring).

**Comment by palkan** [comment]
Hey!

I like the idea of making it possible to invoke other templates. I also usually run a few `rbytes` command when setting up a new project; having just a single one would be useful.

Just to confirm I understood correctly, let me provide a simple example of how it should work.

Under `templates/example-a` I have `template.rb` and `_a.rb.tt` files with the following contents:

```ruby
# template.rb
say "Hello from A template"

<%= include "a" %>

# _a.rb.tt
say "Hello from A partial"
```

Now, in my root template I have the following code:

```ruby
say "Hello from the Root"

<%= import_template "./templates/example-a/template.rb"
```

The resulting compiled template must be equal to:

```ruby
say "Hello from the Root"
say "Hello from A template"
say "Hello from A partial"
```

Is the correct?

Also, as you can see, I propose `import_template` API with just a path to the sub-template entry-point; the root directory is assumed to be the entry-point file's directory (so, no need to specify it).

**Comment by mattdb** [comment]
`import_template` is definitely a better method name - your proposed API makes perfect sense. I'll take at stab at it sometime this week, hopefully. Thanks!

**Comment by palkan** [comment]
Released in 0.2.0
